scenes/cslib_string.txt

```diff
@@ -314,3 +314,61 @@
 		*goto consists_of_char_list_loop
 	*set cslib_ret false
 *return
+
+
+*comment EXTRACT_TOKENS
+*comment ------------------------------
+*comment Split a given string using a separator.
+*comment The tokens are stored in a given array, up to its max.
+*comment ------------------------------
+*comment 		params:
+*comment 			p_source: the string to be split.
+*comment 			p_separator: one character to be used as a separator.
+*comment			p_array(string): the array name/reference (prefix)
+*comment 		returns:
+*comment 			(boolean): the number of tokens in the string.
+*label extract_tokens
+*params p_source p_separator p_array
+
+*set cslib_ret 0
+*set {p_array & "_count"} 0
+
+*temp source_len length(p_source)
+*temp array_max {p_array & "_max"}
+
+*if p_separator = ""
+	*bug Separator should not be empty in extract_tokens.
+*if length(p_separator) > 1
+	*bug Separator should be a single char in extract_tokens.
+
+*comment Clean up the destination array.
```

Comment:
I honestly don't think we should clean the array, for a few of reasons:
- It's unnecessary computation (they get overwritten later anyway)
- Any remaining values are 'excluded' by the setting of _count
- It gives more adventurous users flexibility to retain other data in those now invalid parts

Reply:
It's nice to have them clean so you can directly concatenate into them (no special case for the first char). Also, I believe it's easier to explain that all contents will be nuked (adventurous code is bad code)